outport/notifier/eventNotifier.go

```diff
@@ -0,0 +1,210 @@
+package notifier
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	nodeData "github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/indexer"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+var log = logger.GetOrCreate("outport/eventNotifier")
+
+const (
+	pushEventEndpoint       = "/events/push"
+	revertEventsEndpoint    = "/events/revert"
+	finalizedEventsEndpoint = "/events/finalized"
+)
+
+// SaveBlockData holds the data that will be sent to notifier instance
+type SaveBlockData struct {
+	Hash      string                                 `json:"hash"`
+	Txs       map[string]nodeData.TransactionHandler `json:"txs"`
+	Scrs      map[string]nodeData.TransactionHandler `json:"scrs"`
+	LogEvents []Event                                `json:"events"`
+}
+
+// Event holds event data
+type Event struct {
+	Address    string   `json:"address"`
+	Identifier string   `json:"identifier"`
+	Topics     [][]byte `json:"topics"`
+	Data       []byte   `json:"data"`
+}
+
+// RevertBlock holds revert event data
+type RevertBlock struct {
+	Hash  string `json:"hash"`
+	Nonce uint64 `json:"nonce"`
+	Round uint64 `json:"round"`
+	Epoch uint32 `json:"epoch"`
+}
+
+// FinalizedBlock holds finalized block data
+type FinalizedBlock struct {
+	Hash string `json:"hash"`
+}
+
+type eventNotifier struct {
+	httpClient      httpClientHandler
+	marshalizer     marshal.Marshalizer
+	hasher          hashing.Hasher
+	pubKeyConverter core.PubkeyConverter
+}
+
+// ArgsEventNotifier defines the arguments needed for event notifier creation
+type ArgsEventNotifier struct {
+	HttpClient      httpClientHandler
+	Marshalizer     marshal.Marshalizer
+	Hasher          hashing.Hasher
+	PubKeyConverter core.PubkeyConverter
+}
+
+// NewEventNotifier creates a new instance of the eventNotifier
+// It implements all methods of process.Indexer
+func NewEventNotifier(args ArgsEventNotifier) (*eventNotifier, error) {
+	return &eventNotifier{
+		httpClient:      args.HttpClient,
+		marshalizer:     args.Marshalizer,
+		hasher:          args.Hasher,
+		pubKeyConverter: args.PubKeyConverter,
+	}, nil
+}
+
+// SaveBlock converts block data in order to be pushed to subscribers
+func (en *eventNotifier) SaveBlock(args *indexer.ArgsSaveBlockData) error {
+	log.Debug("SaveBlock called at block", "block hash", args.HeaderHash)
```

Comment:
might add more context on these logs, for example `eventNotifier: SaveBlock called...`

Reply:
added eventNotifier in log message